packages/wallet-sdk/src/sign/interface.ts

```diff
@@ -4,7 +4,7 @@ import { AddressString, Chain } from ':core/type';
 export interface Signer {
   readonly accounts: AddressString[];
   readonly chain: Chain;
-  handshake(): Promise<AddressString[]>;
+  handshake(): Promise<void>;
```

Comment:
we don't need return value from handshake. it's not being used on provider. we only care if it was successful or not. Promise should be enough